# Representação gráfica de funções

Renato Naville Watanabe

## Modelando fenômenos

- Em diferentes áreas da Ciência busca-se modelar fenômenos por médio de funções matemáticas a fim de reproduzir os comportamentos observados na natureza.

- Dado um modelo, muitas vezes, temos a necessidade de visualizar o comportamento do mesmo.

- Gráficos de funções auxiliam o entendimento dos  fenômenos.

![lorentz](https://upload.wikimedia.org/wikipedia/commons/e/ef/Lorenz_Ro28-200px.png)

## Função

- O estudo de funções decorre da necessidade de:
    + Analisar fenômenos, visualizando o comportamento de um sistema.
    + Interpretar interdependências, entendendo como uma variável comporta-se com relação à outra.
    + Encontrar soluções de problemas.
    + Descrever regularidades.
    + Generalizar.

## Definição de uma função

- Uma função é uma regra segundo a qual, para cada elemento $x$ pertencente a um conjunto $A$ corresponde um único elemento $y$ pertencente a um conjunto $B$.

$$y = f(x)$$

- O conjunto $A$ é chamado domínio da função e o conjunto $B$ é o contra-domínio, ou imagem.




## Representação de uma função

- Uma função pode ser representada das seguintes formas: 
    + **Verbalmente** (descrevendo-a com palavras)
    + **Numericamente** (utilizando tabela de valores)
    + **Visualmente** (utilizando gráficos)
    + **Algebricamente** (utilizando uma fórmula explícita)

## Ferramentas de visualização

Existem diversas ferramentas para utilizadas em calculos matemáticos avançados.
- Matlab
- Mathematica
- Octave
- Scilab
- R
- [**Python**](https://nbviewer.jupyter.org/github/BMClab/bmc/blob/master/notebooks/PT-BR/PythonForScientificComputing_PT-BR.ipynb)
- Fortran
- Julia

Geralmente contam com bibliotecas de funções matemáticas prontas e recursos avançados.

## Spyder 
Ambiente de programação Python que permite a criação de novas funções/bibliotecas do usuário.

<img src="imagens/spyder.png" width="400">

O Spyder pode ser obtido com o pacote [Anaconda](https://www.anaconda.com/distribution/).

## Spyder

A interação do usuário com o Spyder pode ocorrer de duas formas distintas:

- Na **primeira forma**, os comando são digitados diretamente no *console* do Spyder:
    + Ao ser pressionada a tecla Enter, os comandos digitados são interpretados e imediatamente executados.
    + O Spyder (Python) funciona como uma  calculadora sofisticada.

- Na **segunda forma**, um conjunto de comandos é digitado em um arquivo texto:
    + Este arquivo  (script) pode ser  executado.
    + Neste modo, o Spyder funciona como um ambiente de programação.

## Spyder

Digitando o seguinte comando no console:

In [4]:
x = 2

estaremos criando uma variável real chamada $x$ cujo valor é igual a 2.

## Variável

Uma variável é um endereço da memória de acesso randômico (RAM), representada por um nome 
(rótulo), criado pelo usuário, cujo conteúdo pode se alterar no decorrer do programa.

Uma variável é composta por dois elementos:
    
  - **Identificador**: nome dado pelo programador à variável
  - **Conteúdo**: valor da variável
<img src="imagens/memvar2.png" width="400">

No caso anterior, x é o identificador (nome da variável) e 2 é o conteúdo (valor).

## Variável

Cada vez que mencionarmos o nome de uma variável, estaremos na verdade utilizando o seu conteúdo

In [16]:
y = x + 5
y

7

Esta operação define $y$ como sendo uma variável com valor igual ao valor de $x$ mais cinco. Ou seja, $y$ tem um valor igual a 7

In [17]:
z = x*y
z

14

Neste caso, $z$ é igual à multiplicação dos valores guardados em $x$ e $y$, ou seja, $z$ é igual a 14.

In [18]:
w = z/x
w

7.0

Aqui, $w$ é igual à divisão dos valores guardados em $z$ e $x$, ou seja, $w$
é igual a 7.

In [19]:
k = w-x
k

5.0

Aqui, $k$ é igual à subtração do valor guardado em $x$ de $w$. Ou seja, $k$
é igual a 5.

In [21]:
m = k**3
m

125.0

Aqui, $m$ é igual ao valor guardado em $k$ elevado ao cubo. Ou seja, $m$ é igual a 125.

In [2]:
import matplotlib.pyplot as plt
import numpy as np

x = np.arange(0, 10, 0.1)
y = x**2
plt.figure()
plt.plot(x,y)
plt.show()


<Figure size 640x480 with 1 Axes>

## Referências

- Chalco, JM, *Slides de Bases Computacionais da Ciência*, (2014)
- Leite, S, *Slides de Bases Computacionais da Ciência*, (2017)
- [Marietto, MGB et al.; **Bases computacionais da Ciência** (2013)](http://prograd.ufabc.edu.br/images/pdf/bases_computacionais_livro.pdf).